comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -412,14 +412,26 @@ func (c *collector) handleServiceRetries(pid int32) {
 }
 
 // getProcessEntitiesFromServices creates Process entities with service discovery data
-func (c *collector) getProcessEntitiesFromServices(newPids []int32, heartbeatPids []int32, pidsToService map[int32]*model.Service) []*workloadmeta.Process {
+func (c *collector) getProcessEntitiesFromServices(newPids []int32, heartbeatPids []int32, pidsToService map[int32]*model.Service, injectedPids core.PidSet) []*workloadmeta.Process {
 	entities := make([]*workloadmeta.Process, 0, len(pidsToService))
 	now := c.clock.Now().UTC()
 
 	// Process new PIDs - create complete entities
 	for _, pid := range newPids {
 		service := pidsToService[pid]
 		if service == nil {
+			// Handle injected processes that are not services
```

Comment:
Are these processes deleted correctly from WLM? AFAICS the deletion from the SD source is done based on the process' presence in the `pidHeartbeats` maps, but these processes are never added there.